### PR TITLE
[WIP] Adding auth to binder

### PR DESCRIPTION
**Pull request by jhamman**
Following @scottyhq's lead, this adds github auth to pangeo's binder deployment. N

### CODE REVIEW

File: pangeo-binder/values.yaml [hunk 1]
```diff
@@ -64,7 +72,26 @@ binderhub:
...
           cpu: "1.25"
           memory: 1Gi
+      allowNamedServers: true
+      # change this value as you wish,
+      # or remove this line if you don't want to have any limit
+      # namedServerLimitPerUser: 5
+      redirectToServer: false
+      services:
+        binder:
+          oauth_redirect_uri: "hhttps://hub.staging.binder.pangeo.io/hub/oauth_callback"
```
Comment (scottyhq) [inline pangeo-binder/values.yaml]: @jhamman - i think you just need to remove the extra 'h' here

File: pangeo-binder/values.yaml [hunk 2]
```diff
@@ -27,6 +29,10 @@ binderhub:
           kubernetes.io/ingress.class: nginx
           kubernetes.io/tls-acme: 'true'
     cull:
+      # don't cull authenticated users
+      users: False
+      namedServerLimitPerUser: 5
+      redirectToServer: false
```
Comment (bitnik) [inline pangeo-binder/values.yaml]: These 2 configs (`namedServerLimitPerUser` and `redirectToServer`) have no effect here, they should be under `jupyterhub.hub`

File: pangeo-binder/values.yaml [hunk 3]
```diff
@@ -64,7 +72,26 @@ binderhub:
...
           cpu: "1.25"
           memory: 1Gi
+      allowNamedServers: true
+      # change this value as you wish,
+      # or remove this line if you don't want to have any limit
+      # namedServerLimitPerUser: 5
+      redirectToServer: false
+      services:
+        binder:
+          oauth_redirect_uri: "https://hub.staging.binder.pangeo.io/hub/oauth_callback"
```
Comment (bitnik) [inline pangeo-binder/values.yaml]: "https://hub.staging.binder.pangeo.io/hub/oauth_callback" -> I think this should be url of the binderhub: "https://staging.binder.pangeo.io/hub/oauth_callback"

File: pangeo-binder/values.yaml (hunk 2 above)
Comment (bitnik) [reply pangeo-binder/values.yaml]: Note: there is a bug with `namedServerLimitPerUser` config (https://github.com/jupyterhub/jupyterhub/issues/2758). It is fixed by https://github.com/jupyterhub/jupyterhub/pull/2772 but it is not released yet.

**Comment by jhamman** [comment]
Thanks @bitnik for the review. Things are working now: https://staging.binder.pangeo.io.

**Comment by jhamman** [comment]
This seems to be working quite well. While I love the functionality this offers, I have two UX 

1. Users get an uninformative error from binderhub when they bump into `namedServerLimitPerUser`. I've turned this off for now. 
2. Having to authenticate before hitting the binder landing page  is a bit obtrusive. I wonder if this is something that we can work to improve. Maybe this just means adding some css/html to the JupyterHub auth pages. Not sure the best way to proceed here.

**Comment by bitnik** [comment]
> Users get an uninformative error from binderhub when they bump into namedServerLimitPerUser. I've turned this off for now.

There is now an issue about this: https://github.com/jupyterhub/binderhub/issues/1022

**Pull request merged**